cloudify_ansible/utils.py

```diff
@@ -162,18 +165,37 @@ def handle_sources(data, site_yaml_abspath, _ctx):
     return hosts_abspath
 
 
-def handle_source_from_string(filepath, _ctx, new_inventory_path):
-    if '\n' not in filepath:
-        # We are assuming that there is no file path with \n in it.
-        # We are also assuming that there are multiple lines in an
-        # inventory file :(
+def get_inventory_file(filepath, _ctx, new_inventory_path):
+    """
+    This method will get the location for inventory file.
+    The file location could be locally with relative to the blueprint
+    resources or it could be remotely on the remote machine
+    :return:
+    :param filepath: File path to do check for
+    :param _ctx: The Cloudify context.
+    :param new_inventory_path: New path which holds the file inventory path
+    when "filepath" is a local resource
+    :return: File location for inventory file
+    """
+    if os.path.isfile(filepath):
+        # The file already exists on the system, then return the file url
+        return filepath
+    else:
+        # Check to see if the file does not exit, then try to lookup the
+        # file from the Cloudify blueprint resources
         try:
-            open(filepath, 'r')
-            # The file already exists on the system.
-            return filepath
-        except IOError:
-            # The file doesn't exist on the system, so we need to download it.
             _ctx.download_resource(filepath, new_inventory_path)
+        except HttpException:
+            _ctx.logger.error(
+                'Error when trying to download {0}'.format(filepath))
+            return None
+        return new_inventory_path
+
+
+def handle_source_from_string(filepath, _ctx, new_inventory_path):
+    inventory_file = get_inventory_file(filepath, new_inventory_path)
+    if inventory_file:
+        new_inventory_path = inventory_file
```

Comment:
i think 197-198 are redundant because inventory_file is already pointing to new_inventory_path if it is None. So you can just use inventory_file for the rest of the method.

Reply:
@EarthmanT `inventory_file` here could have one of the following values:
1. `filepath`
2. `new_inventory_path`
3. `None`

If its `None` that means this is not a valid input file and we need to dump the content to the `new_inventory_path` otherwise then we need to assign  `new_inventory_path` to `inventory_file` because it could be a remote file located on the system

Reply:
but because you pass `new_inventory_file` to the `get_inventory_file` function, if the value is `new_inventory_file` then so is `inventory_file`. So you do not need to reassign the value.

Reply:
You are right but re-assigning is required when `inventory_file` is not the same as `new_inventory_file` so this code can be updated to the following:
```
    inventory_file = get_inventory_file(filepath, _ctx, new_inventory_path)
    if inventory_file and inventory_file != new_inventory_path:
        new_inventory_path = inventory_file
``
@EarthmanT what do you think